uniffi_bindgen/src/bindings/gecko_js/webidl.rs

```diff
@@ -50,6 +50,13 @@ pub enum WebIDLType {
     Nullable(Box<WebIDLType>),
     Optional(Box<WebIDLType>),
 
+    /// Optionals with a default value are a grab bag of special cases in Gecko.
+    /// In the generated C++ bindings, the type of an optional with a default
+    /// value is `T`, not `Optional<T>`. However, it must be serialized as if
+    /// it's an `Optional<T>`, since that's what the Rust side of the FFI
+    /// expects.
```

Comment:
What a headache, I wonder if we shouldn't have disallowed optionals without default values in IDL files.